spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplate.java

```diff
@@ -818,19 +817,30 @@ private DatastoreReaderWriter getDatastoreReadWriter() {
 		DatastorePersistentEntity<?> persistentEntity =
 				this.datastoreMappingContext.getPersistentEntity(example.getProbeType());
 
-		StructuredQuery.Builder builder = keyQuery ? Query.newKeyQueryBuilder() : Query.newEntityQueryBuilder();
-		builder.setKind(persistentEntity.kindName());
-
-		ExampleMatcherAccessor matcherAccessor = new ExampleMatcherAccessor(example.getMatcher());
-		matcherAccessor.getPropertySpecifiers();
 		LinkedList<StructuredQuery.Filter> filters = new LinkedList<>();
+		NullHandler nullHandler = example.getMatcher().getNullHandler();
 		persistentEntity.doWithColumnBackedProperties((persistentProperty) -> {
 			if (!ignoredProperty(example, persistentProperty)) {
 				Value<?> value = getValue(example, probeEntity, persistentEntity, persistentProperty);
-				NullHandler nullHandler = example.getMatcher().getNullHandler();
 				addFilter(nullHandler, filters, persistentProperty.getFieldName(), value);
 			}
 		});
+		persistentEntity.doWithAssociations((AssociationHandler<DatastorePersistentProperty>) association -> {
```

Comment:
Unnecessary cast?

Reply:
this is necessary, otherwise there is ambiguous call error